File: app/src/main/cpp/GameElement/GameObject/StrongBrick.cpp
#include <android/log.h>

#include "StrongBrick.h"

#define TAG "StrongBrickCpp"
#define LOGD(...) __android_log_print(ANDROID_LOG_DEBUG, TAG, __VA_ARGS__)

StrongBrick::StrongBrick(int row, int column, int width, int height)
        : Brick(row, column, width, height) {
    maxHits = 3;
    updateColor();
}

void StrongBrick::hit() {
    hitCount++;
    if (hitCount >= maxHits) {
        setInvisible();
    } else {
        updateColor();
    }
    LOGD("StrongBrick hit - hitCount: %d", hitCount);
}

void StrongBrick::updateColor() {
    switch (hitCount) {
        case 0:
            currentColor = 0xFFFF0000;  // RED
            break;
        case 1:
            currentColor = 0xFFFFFF00;  // YELLOW
            break;
        case 2:
            currentColor = 0xFF00FF00;  // GREEN
            break;
    }
}

File: app/src/main/java/com/example/breakoutgame/GameElement/GameObject/StrongBrick.java
package com.example.breakoutgame.GameElement.GameObject;

import android.graphics.Color;

public class StrongBrick extends Brick {

    public StrongBrick(int row, int column, int width, int height) {
        super(row, column, width, height);
        this.maxHits = 3; // Strong brick breaks with three hits
        updateColor();
    }

    @Override
    protected void updateColor() {
        switch (hitCount) {
            case 0:
                currentColor = Color.RED;
                break;
            case 1:
                currentColor = Color.YELLOW;
                break;
            case 2:
                currentColor = Color.GREEN;
                break;
        }
        nativeSetColor(nativePtr, currentColor);
    }
}
